File: C++/binary_tree_inorder_traversal.cpp
// no. 94
#include "tree_node.h"
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> result;
  void helper(TreeNode *root) {
    if (!root) {
      return;
    }
    if (root->left) {
      helper(root->left);
    }
    result.push_back(root->val);
    if (root->right) {
      helper(root->right);
    }
  }
  vector<int> inorderTraversal(TreeNode *root) {
    helper(root);
    return result;
  }
};

File: python/binary_tree_inorder_traversal.py
# no. 94
from utils import TreeNode

# left, middle, right
class Solution:
    def __init__(self):
        self.result = []
    
    def helper(self, root: Optional[TreeNode]):
        if not root:
            return
        if root.left:
            self.helper(root.left)
        self.result.append(root.val)
        if root.right:
            self.helper(root.right)
        
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.helper(root)
        return self.result
